oneview/resource_server_profile.go

```diff
@@ -35,14 +35,18 @@ func resourceServerProfile() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
 			"template": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
 			},
 			"type": {
 				Type:     schema.TypeString,
 				Optional: true,
-				Default:  "ServerProfileV5",
+				Default:  "ServerProfileV9",
```

Comment:
This will cause a change if someone was using `ServerProfileV5` and then upgrades to this version of the provider and now Terraform wants to upgrade their type to `ServerProfileV9`. Is that what you want users to do? You'll want to think of a story for upgrading if this is the route you want to go

Reply:
Instead of a Default you probably should just have users validate against a list of types
```
ValidateFunc: validation.StringInSlice([]string{
  "ServerProfileV5",
  "ServerProfileV9",
}, true)
```

Reply:
The type field is a Required attribute for the POST request. Can we have both a default value and a validate function as part of the schema?
Or would it be better to make the field required and add the validate function to it? But even in this case, the user when he upgrades would have to provide this value explicitly if he was not doing so before.

Reply:
@mbfrahry, we had a discussion within the team. We would be adding the validate function as part of code refactoring in the future. I'm reverting the type to the old value as of now.